pkg/ruler/store_mock_test.go

```diff
@@ -131,6 +134,21 @@ func (m *mockRuleStore) ListAllRuleGroups(_ context.Context) (map[string]rulespb
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
 
+	if m.loaded {
```

Comment:
Can we simply build the result on each call, without changing internal state of `mockRuleStore`? I find `m.loaded` confusing, and not sure what it is trying to express.

In other words, I would suggest to remove `m.loads` and `m.loaded`, and just build the result on each call to `ListAllRuleGroups` ("unloaded"). Same for `ListRuleGroupsForUserAndNamespace`. WDYT?

Reply:
If we want change `mockRuleStore` to require LoadRuleGroups, then we should not return `Rules` in `ListAllRuleGroups`, and fill `Rules` in `LoadRuleGroups `. 

The problem is that in `mockRuleStore` we have no other storage to store that `Rules`.

Reply:
Idea was that `ListAllRuleGroups` would build the result, keeping only `Namespace`, `Name` and `User` fields. Similar to what your code already does, but without extra fields in `mockRuleStore`.

Reply:
Get it, thanks.